cmd/clusterctl/cmd/topology.go

```diff
@@ -0,0 +1,306 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"sort"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/cluster"
+	utilyaml "sigs.k8s.io/cluster-api/util/yaml"
+)
+
+type topologyDryRunOptions struct {
+	file    string
+	cluster string
+	outDir  string
+}
+
+var dr = &topologyDryRunOptions{}
+
+var topologyDryRunCmd = &cobra.Command{
+	Use:   "topology-dryrun",
+	Short: "Dry run changes to clusters that use managed topologies.",
+	Long: LongDesc(`
+	    Provide the list of objects that would be created, modified and deleted when a change is applied.
```

Comment:
Oh great! I got the impression it only dry-runs changes to clusters. Maybe you can double-check the command description if it's already clear enough, but it could absolute be just me.

Feel free to resolve

Reply:
I think it was
"Dry run **changes** to clusters that use managed topologies"
and
"List the changes when **modifying** a cluster."
that gave me the idea

Reply:
I modified some wording and added a few more examples. Hopefully it will make it clear.